### DISCUSSION TITLE
Group ingredients for steps

### DISCUSSION THREAD

**Discussion by MarJMue** [mealie-recipes/mealie#4682] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Ingredient lists for recipes with many steps and ingredients can get difficult to read. I would like the ability to add further partition into steps.

### Please provide a concise description of the feature that would resolve your issue.

I quite like the style of the cuisine latex package and some textbooks I own, to group the ingredients for each step.
Like in this example:
![Screenshot_20241205_091323](https://github.com/user-attachments/assets/739f9f53-7382-4956-bc0d-8b8ba36126e7)

### Please consider and list out some caveats or tradeoffs made in your design decision

I unfortunately have no idea how difficult it may be to implement.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
Hey, i think we already have all the features you are asking about in mealie. 

1. You can create sections for ingredients (as well as steps) via the three dot menu which will give you groups that are easier to handle 
2. You can go one step further and link ingredients to steps (three dot menu on a step) which will  result in something like the image below when you switch to cookmode.
  2.1. there is one limitation that you are not able to use half of the ingredients in step a and half in step b, but you are able to link the same ingredient to multiple step so you can mitigate that by either having two times the same ingredient in the recipe so you can link it, or noting down in the step that this step only uses half of the ingredient 
---

<img src="https://github.com/user-attachments/assets/b5477082-82e1-4db8-bbcc-108d18006a6f"  width="500px">

**Reply by MarJMue** [reply]
Hey, thank you. That looks exactly as I would like. I got nearly all the way to your picture. 
It's ok in the normal mode, but cooking-mode does append the ingredients below the instructions for me.
![image](https://github.com/user-attachments/assets/3f4f4166-70c4-40c6-812d-4b6a885511f9)

**Reply by MarJMue** [reply]
Nevermind. This was on 2.2.0. I completely missed version 2.3.0
Works perfectly. :+1:
